webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/data/NodeDetailLevel.scala

```diff
@@ -159,61 +161,63 @@ object NodeDetailLevel {
     )
   }
 
-  private val nodeInfoFields: Map[String, NodeInfo => JValue] = {
-
-    val id           : NodeInfo => JValue = (inv: NodeInfo) => inv.id.value
-    val hostname     : NodeInfo => JValue = (inv: NodeInfo) => inv.hostname
-    val state        : NodeInfo => JValue = (inv: NodeInfo) => inv.state.name
-    val description  : NodeInfo => JValue = (inv: NodeInfo) => inv.description
-    val policyServer : NodeInfo => JValue = (inv: NodeInfo) => inv.policyServerId.value
-    val ram          : NodeInfo => JValue = (inv: NodeInfo) => inv.ram.map(MemorySize.sizeMb)
-    val arch         : NodeInfo => JValue = (inv: NodeInfo) => inv.archDescription
-    val inventoryDate: NodeInfo => JValue = (inv: NodeInfo) => DateFormaterService.getFormatedDate(inv.inventoryDate)
-    val properties   : NodeInfo => JValue = (inv: NodeInfo) => inv.properties.toApiJson
-    val policyMode   : NodeInfo => JValue = (inv: NodeInfo) => inv.policyMode.map(_.name).getOrElse[String]("default")
-    val timezone     : NodeInfo => JValue = (inv: NodeInfo) => inv.timezone.map( t => ("name" -> t.name) ~ ("offset" -> t.offset) )
+  type INFO = (NodeInfo, Option[DateTime])
+  private val nodeInfoFields: Map[String, INFO => JValue] = {
+
+    val id           : INFO => JValue = (info:INFO) => info._1.id.value
+    val hostname     : INFO => JValue = (info:INFO) => info._1.hostname
+    val state        : INFO => JValue = (info:INFO) => info._1.state.name
+    val description  : INFO => JValue = (info:INFO) => info._1.description
+    val policyServer : INFO => JValue = (info:INFO) => info._1.policyServerId.value
+    val ram          : INFO => JValue = (info:INFO) => info._1.ram.map(MemorySize.sizeMb)
+    val arch         : INFO => JValue = (info:INFO) => info._1.archDescription
+    val runDate      : INFO => JValue = (info:INFO) => info._2.map(d => JString(DateFormaterService.getFormatedDate(d))).getOrElse(JNothing)
```

Comment:
So here you won't have "lastRunDate" field if you ask for it and there was no run present, maybe we should always have a value, saying 'never' ? what do you think of this ?

Reply:
I wasn't sure. It seems easier/more idiomatic for user to test for "field is null" than for an arbitrary value. But really, I'm not sure of idiomatism for these thing

Reply:
"idiotmatism"  would be fine for JSON ;)

What do our team members thinks ? what do people thinks ? what does the internet say ? let's look !